### PR TITLE
V1.1

### PR DESCRIPTION
**Pull request by jeffherman**
Fixes and implementations for several issues.

- Update docstrings to Google style (#7)
- Fixes for Python 3.0+ (#8)
- Fix macOS version parsing (#9)
- Return proper types for each field (#10)
- Return data as named tuple (#11)

### CODE REVIEW

**Review by tuvwxy — APPROVED** [review]

File: usbinfo/__init__.py [hunk 1]
```diff
@@ -127,13 +127,84 @@
...
+
+def endpoints(**filters):
+    """Return a list of named tuples describing attached USB endpoints.
+
+    This function is similar to ``usbinfo()`` but returns a list of named tuples
+    for more Pythonic access to information on each endpoint.
+
+    Args:
+        **filters: A key-value list of filters where the value type is one of
+            `str`, `int`, or a compiled regular expression.
```
Comment (yusufm) [inline usbinfo/__init__.py]: It looks like None is acceptable as well (given the conditional in key_value_match()?

File: usbinfo/__main__.py [hunk 2]
```diff
@@ -29,17 +28,21 @@
...
 
 def _max_width(string_list):
-    """
-    Helper to return the width of the longest string.
+    """Helper to return the width of the longest string.
+
+    Args:
+        string_list: List of strings to search.
     """
     return max([len(item) for item in string_list])
```
Comment (yusufm) [inline usbinfo/__main__.py]: This should work instead of a new func.

max(stringlist, key=len)

File: usbinfo/__main__.py [hunk 3]
```diff
@@ -178,7 +190,7 @@ def __init__(self):
 
 
 def main(args=None):
-    """Main function"""
+    """Main function."""
```
Comment (yusufm) [inline usbinfo/__main__.py:181]: Can take this comment out.

File: usbinfo/__init__.py [hunk 4]
```diff
@@ -127,13 +127,84 @@
...
+    Args:
+        ep_dict: Dictionary object representing each USB endpoint.
+    Returns:
+        An Endpoint object representing the input dictionary.
     """
+    def cast_int(key):
+        """Cast value of key to int or fallback to None."""
+        value = ep_dict.get(key, '')
+        if value.isdigit():
+            return int(value, 16)
```
Comment (yusufm) [inline usbinfo/__init__.py:151]: nit: I'd prefer explicitly returning None at the end.

File: usbinfo/__main__.py (hunk 3 above)
Comment (jeffherman) [reply usbinfo/__main__.py:181]: Probably will leave this in for style-guide conformity.

**Review by yusufm — APPROVED** [review]

**Pull request merged**